js/plugins/mcp/src/util/resources.ts

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type {
+  Resource,
+  ResourceTemplate,
+} from '@modelcontextprotocol/sdk/types.js';
+import { Genkit, z } from 'genkit';
+
+function listResourcesTool(
+  ai: Genkit,
+  client: any, // Use 'any' or let TS infer; removing specific type import
```

Comment:
Why any? If it's a circular dependency issue, you can do `import type` without triggering circ dependencies.